src/Compilers/CSharp/Portable/Symbols/Source/SourceDelegateClonedParameterSymbolForBeginAndEndInvoke.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SourceDelegateClonedParameterSymbolForBeginAndEndInvoke : SourceClonedParameterSymbol
+    {
+        internal SourceDelegateClonedParameterSymbolForBeginAndEndInvoke(SourceParameterSymbol originalParam, SourceDelegateMethodSymbol newOwner, int newOrdinal)
+            : base(originalParam, newOwner, newOrdinal, suppressOptional: true)
+        {
+        }
+
+        internal override bool IsCallerFilePath => _originalParam.IsCallerFilePath;
```

Comment:
I didn't understand why properties that are implemented the same way in `SourceDelegateClonedParameterSymbolForBeginAndEndInvoke` and `SourcePropertyClonedParameterSymbolForAccessors` were moved out of the base type.

Reply:
@jcouv This was requested by @AlekseyTs. The reasoning is to *force* new types to implement it. i.e, avoid future mistakes if a new type should implement it another way.